### PR TITLE
Feature/50 service user questions

### PR DESCRIPTION
**Pull request by denny13**
Adding service users component between service capacity and data shares.

### CODE REVIEW

File: src/app/features/select-main-service/select-main-service.component.ts [hunk 1]
```diff
@@ -86,7 +86,10 @@ export class SelectMainServiceComponent implements OnInit {
 
     this._registrationService.getMainServices(this.regulatedCheck)
       .subscribe(
-        value => this.allCategoriesArray = value
+        value => {
+          debugger;
```
Comment (wozzer72) [inline src/app/features/select-main-service/select-main-service.component.ts]: remove all debugger statements

Comment (wozzer72) [reply src/app/features/select-main-service/select-main-service.component.ts]: There should be no change on this component once the debugger statement has been removed

File: src/app/features/service-users/service-users.component.ts [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+  public isInvalid: boolean;
+  public serviceUsersData = [];
+  public checkboxesSelected;
+  private subscriptions: Subscription = new Subscription();
+
+  constructor(
+    private fb: FormBuilder,
+    private router: Router,
+    private route: ActivatedRoute,
+    private _eSService: EstablishmentService
```
Comment (ChazUK) [inline src/app/features/service-users/service-users.component.ts]: no need to use _, should be named `establishmentService`

File: src/app/features/service-users/service-users.component.ts [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+            if (data.serviceUsers) {
+              data.serviceUsers.forEach(thisServiceUser => {
+                this.checkboxesSelected.push(thisServiceUser.id);
+              });
+            }
+          },
+          (err) => {
+            console.log(err);
+          },
+          () => {
```
Comment (ChazUK) [inline src/app/features/service-users/service-users.component.ts]: remove blank complete function

File: src/app/features/service-users/service-users.component.ts [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+  }
+
+  toggleCheckbox($event: any) {
+    const serviceUserId = $event.value;
+
+    if ($event.checked) {
+      // add the serviceId to the known set of selected checkbox; but opnly if it
+      //  doesn't already exist
+      if (!this.checkboxesSelected.includes(serviceUserId)) {
+        this.checkboxesSelected.push(parseInt(serviceUserId));
```
Comment (ChazUK) [inline src/app/features/service-users/service-users.component.ts]: should be `parseInt(var, 10)`

File: src/app/features/select-main-service/select-main-service.component.ts (hunk 1 above)
Comment (denny13) [reply src/app/features/select-main-service/select-main-service.component.ts]: Done... dang thought i'd got them all lol

File: src/app/features/service-users/service-users.component.ts [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+      }
+    } else {
+      // remove the given service id
+      const foundServiceIdIndex = this.checkboxesSelected.indexOf(parseInt(serviceUserId));
```
Comment (ChazUK) [inline src/app/features/service-users/service-users.component.ts]: should be `parseInt(var, 10)`

File: src/app/features/service-users/service-users.component.ts [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+        this.checkboxesSelected.splice(foundServiceIdIndex, 1);
+      }
+    }
+  }
+
+  async onSubmit() {
+    const serviceUsersSelected = {
+      serviceUsers: this.checkboxesSelected.map(thisValue => {
+          return {
+            id: parseInt(thisValue)
```
Comment (ChazUK) [inline src/app/features/service-users/service-users.component.ts]: should be `parseInt(var, 10)`

File: src/app/features/service-users/service-users.component.ts (hunk 2 above)
Comment (denny13) [reply src/app/features/service-users/service-users.component.ts]: Just used what is currently being used in establishments.. but sure can update.

Comment (denny13) [reply src/app/features/service-users/service-users.component.ts]: Done

File: src/app/features/service-users/service-users.component.ts (hunk 3 above)
Comment (denny13) [reply src/app/features/service-users/service-users.component.ts]: Done

File: src/app/features/service-users/service-users.component.ts (hunk 4 above)
Comment (denny13) [reply src/app/features/service-users/service-users.component.ts]: Done

File: src/app/features/service-users/service-users.component.ts (hunk 5 above)
Comment (denny13) [reply src/app/features/service-users/service-users.component.ts]: Done

File: src/app/features/service-users/service-users.component.ts (hunk 6 above)
Comment (denny13) [reply src/app/features/service-users/service-users.component.ts]: Done

**Comment by ChazUK** [comment]
I agree with Warren that the navigation should be fixed before this is merged.

**Pull request merged**